packages/editor/src/components/browser/TypeBrowser.tsx

```diff
@@ -13,40 +14,99 @@ export const useTypeBrowser = (onDoubleClick: () => void): UseBrowserImplReturnV
     id: TYPE_BROWSER_ID,
     name: 'Type',
     content: <TypeBrowser value={value} onChange={setValue} onDoubleClick={onDoubleClick} />,
-    accept: () => value
+    accept: () => value,
+    icon: IvyIcons.DataClass
   };
 };
 
+type TypeKind = 'DATACLASS' | 'NORMALTYPE' | 'OWNTYPE';
+
 const TypeBrowser = (props: { value: string; onChange: (value: string) => void; onDoubleClick: () => void }) => {
+  type TypeBrowserObject = JavaType & {
+    kind: TypeKind;
+  };
+
   const { context } = useEditorContext();
 
+  const [allSearchActive, setAllSearchActive] = useState<boolean>(false);
+
   const [mainFilter, setMainFilter] = useState('');
   const { data: allDatatypes, isFetching } = useMeta('meta/scripting/allTypes', { context, limit: 150, type: mainFilter }, []);
+  const dataClasses = useMeta('meta/scripting/dataClasses', context, []).data;
+  const ivyTypes = useMeta('meta/scripting/ivyTypes', undefined, []).data;
 
-  const [dynamicTypes, setDynamicTypes] = useState<JavaType[]>([]);
+  const [types, setTypes] = useState<TypeBrowserObject[]>([]);
 
   const [typeAsList, setTypeAsList] = useState(false);
 
   const [showHelper, setShowHelper] = useState(false);
 
   useEffect(() => {
-    if (mainFilter.length > 0) {
-      allDatatypes.sort((a, b) => a.simpleName.localeCompare(b.simpleName));
+    if (allSearchActive) {
+      if (mainFilter.length > 0) {
+        allDatatypes.sort((a, b) => a.simpleName.localeCompare(b.simpleName));
+      }
+      const mappedAllTypes: TypeBrowserObject[] = allDatatypes.map<TypeBrowserObject>(type => ({
+        kind: 'NORMALTYPE',
+        ...type
+      }));
+      setTypes(mappedAllTypes);
+    } else {
+      const mappedDataClasses: TypeBrowserObject[] = dataClasses.map<TypeBrowserObject>(dataClass => ({
+        simpleName: dataClass.name,
+        kind: 'DATACLASS',
+        ...dataClass
+      }));
+      const mappedIvyTypes: TypeBrowserObject[] = ivyTypes.map<TypeBrowserObject>(ivyType => ({
+        kind: 'NORMALTYPE',
+        ...ivyType
+      }));
+
+      const sortedIvyTypes = mappedIvyTypes.slice().sort((a, b) => {
+        const fqCompare = a.fullQualifiedName.localeCompare(b.fullQualifiedName);
+        if (fqCompare !== 0) {
+          return fqCompare;
+        }
+        return a.simpleName.localeCompare(b.simpleName);
+      });
+
+      const sortedMappedDataClasses = mappedDataClasses.slice().sort((a, b) => {
+        const fqCompare = a.fullQualifiedName.localeCompare(b.fullQualifiedName);
+        if (fqCompare !== 0) {
+          return fqCompare;
+        }
+        return a.simpleName.localeCompare(b.simpleName);
+      });

```

Comment:
deduplicate this compare logic into a shared function.
A cool improvement, but I'm not sure if the backend shouldn't already deliver this types sorted. But ok for now to leave this code here in the client :)